### PR TITLE
Redis4

### PR DESCRIPTION
**Pull request by oleurud**
Solves #819 

Updating `node-redis` package adding Redis 4 support. We must change:
- `node-cartodb-redis` https://github.com/CartoDB/node-cartodb-redis/pull/19, 
- `redis-mpool` https://github.com/CartoDB/node-redis-mpool/pull/9 
- `windshaft` https://github.com/CartoDB/Windshaft/pull/588

Of course, this change must be work with Redis 3.

### CODE REVIEW

**Comment by oleurud** [comment]
It is not a production ready version (I need to create and publish npm versions of `Windshaft` and `redis-mpool`), but it has the code.

**Comment by oleurud** [comment]
Doing manual tests of this PR, ensuring the good behavior with `CL.THROTTLE` (or any other Redis command), I have found an important key.

Using a Lua script, it is not necessary to upgrade the `node-redis` version. The Lua script won't check anything, the Redis server receives the command and parameters and executes them.

So, we are not obligated to upgrade `node-redis`, beacuse we are going to use a Lua script (we need to execute 2 command in Redis)

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "camshaft": "0.60.0",
     "cartodb-psql": "0.10.2",
     "cartodb-query-tables": "0.3.0",
-    "cartodb-redis": "0.14.0",
+    "cartodb-redis": "cartodb/node-cartodb-redis#redis4",
```
Comment (dgaubert) [inline package.json]: Let's wait for `node-cartodb-redis` to be released before proceeding.

File: package.json [hunk 2]
```diff
@@ -40,14 +40,14 @@
     "node-statsd": "~0.0.7",
     "on-headers": "^1.0.1",
     "queue-async": "~1.0.7",
-    "redis-mpool": "0.4.1",
+    "redis-mpool": "cartodb/node-redis-mpool#redis4",
```
Comment (dgaubert) [inline package.json]: Let's wait for `node-redis-mpool` to be released before proceeding.

File: package.json [hunk 3]
```diff
@@ -40,14 +40,14 @@
...
     "request": "^2.83.0",
     "semver": "~5.3.0",
     "step": "~0.0.6",
     "step-profiler": "~0.3.0",
     "turbo-carto": "0.20.2",
     "underscore": "~1.6.0",
-    "windshaft": "4.3.3",
+    "windshaft": "cartodb/windshaft#redis4",
```
Comment (dgaubert) [inline package.json]: Let's wait for `windshaft` to be released before proceeding.

**Comment by oleurud** [comment]
All is ready, how do you see it @dgaubert ?

**Pull request merged**